packages/stream-receiver/README.md

```diff
@@ -0,0 +1,297 @@
+
+First, create a **[`StreamServer`](#StreamServer)** with the base ILP address of the STREAM server and previously generated server secret:
+
+```js
+import { StreamServer } from '@interledger/stream-receiver'
+
+const server = new StreamServer({
+  serverSecret: Buffer.from(PROCESS.env.SERVER_SECRET, 'hex'), // Example: '61a55774643daa45bec703385ea6911dbaaaa9b4850b77884f2b8257eef05836'
+  serverAddress: PROCESS.env.SERVER_ADDRESS, // Example: 'g.mywallet.receiver'
+})
+```
+
+Then, for the proper endpoints within the server (this snippet uses [Express](https://expressjs.com/)), generate and return a new set of connection credentials:
+
+```js
+express().get('/.well-known/open-payments', (req, res) => {
+  const credentials = server.generate()
+  return req.json({
+    ilpAddress: credentials.ilpAddress,
+    sharedSecret: credentials.sharedSecret.toString('base64'),
+  })
+})
+```
+
+The credentials include a unique ILP address to identify this connection, and a shared encryption key so the client can encrypt STREAM messages so other connectors cannot read or tamper with them. The encryption key also enables them to generate conditions for ILP Prepare packets that this STREAM server can fulfill. Note: these credentials are only valid when used with this STREAM server implementation, and not `ilp-procotol-stream`, or the Java or Rust implementations.
+
+To support [STREAM receipts](https://interledger.org/rfcs/0039-stream-receipts/), a feature that enables sender to prove to a third-party verifier how much has been delivered, input the nonce and secret from the request when generating credentials:
+
+```js
+server.generateCredentials({
+  receiptSetup: {
+    nonce: Buffer.from(req.headers['Receipt-Nonce'], 'base64'),
+    secret: Buffer.from(req.headers['Receipt-Secret'], 'base64'),
+  },
+})
+```
+
+If generating credentials for an [Open Payments invoice](https://docs.openpayments.dev/invoices), the operator could encode necessary metadata into the generated credentials using the `paymentTag` option, such as an invoice ID:
+
+```js
+server.generateCredentials({
+  paymentTag: 'a6bbd8e4-864a-4e52-b037-7938e00e6537',
+})
+```
+
+The `paymentTag` will be exposed on each incoming packet so the operator can correlate it with the correct user. The operator can choose any format or data to encode into the `paymentTag`, so long as it's limited to ASCII characters. These details will be securely encrypted into the ILP address, so neither the sender nor any other connectors can read them.
+
+Lastly, only when generating credentials for an [SPSP](https://interledger.org/rfcs/0009-simple-payment-setup-protocol/) request (but not for Open Payments), the operator should provide the asset and denomination of the recipient, which will later be shared with the client over STREAM:
```

Comment:
I could be convinced, but I think I'd prefer to leave the API a little bit more generic for now. Point well taken that there's a potential footgun if an asset isn't provided when used with SPSP, but that should be apparent since some payments will fail.